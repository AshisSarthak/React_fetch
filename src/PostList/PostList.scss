.post-list {
    width: 50%;
    max-height: 400px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: 1px 0px 2px #aaa;
    overflow: auto;
    margin: 2% 25%;
    cursor: pointer;
}