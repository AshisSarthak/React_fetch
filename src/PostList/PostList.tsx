import React, { ReactNode } from 'react';
import PostItemEl from './PostItem/PostItem';
import "./PostList.scss";
import { UserDetails } from '../User/User';

export interface PostItem {
  userId: number;
  id: number;
  title: string;
  body: string;
}

interface PostListState {
  postList: PostItem[];
}
interface PostListProp {
  selectUser: (user:UserDetails) => void;
  selectPost: (post:PostItem) => void;
}
export class PostList extends React.Component<PostListProp, PostListState> {

  constructor(props:PostListProp, state:PostListState){
    super(props, state);
    this.state = {
      postList:[]
    };
  }

  componentDidMount(){
    fetch('https://jsonplaceholder.typicode.com/posts')
  .then(response => response.json())
  .then(json => this.setState({
    postList: json
  }));
  }

  generatePost():ReactNode {
    return this.state.postList.map((item, index)=>{
      return(
        <PostItemEl
          key={index}
          item={item}
          selectPost={this.props.selectPost}
          selectUser={this.props.selectUser}
        />
      )
    })
  }

  render() {
    return (
      <section className="post-list">
        {
          this.generatePost()
        }
      </section>
    );  
  }
}

export default PostList;
