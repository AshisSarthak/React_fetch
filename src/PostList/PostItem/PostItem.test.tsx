import React from "react";
import { shallow } from "enzyme";
import fetchMock from "jest-fetch-mock";
import PostItemEl from "./PostItem";
import { PostItem } from "../PostList";

const demoPost: PostItem = {
  body: "demo",
  id: 1,
  title: "demo",
  userId: 11,
};

describe("PostItem Componnet", () => {
  afterEach(() => {
    fetchMock.resetMocks();
  });

  it("default render", () => {
    const selectuserMock = jest.fn();
    const selectPostMock = jest.fn();
    const wrapper = shallow(
      <PostItemEl
        item={demoPost}
        selectPost={selectPostMock}
        selectUser={selectuserMock}
      />
    );
    expect(wrapper).toMatchSnapshot();

    const contentEl = wrapper.find("div.content");
    contentEl.simulate("click");
    expect(selectPostMock).toBeCalledWith(demoPost);
  });
});
