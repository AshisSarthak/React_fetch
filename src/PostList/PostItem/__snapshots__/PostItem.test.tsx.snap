// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem Componnet default render 1`] = `
<div
  className="post-item"
>
  <div
    className="content"
    onClick={[Function]}
  >
    demo
  </div>
</div>
`;
